Controllers/Backend/ConnectGateway.php

```diff
@@ -134,11 +134,33 @@ public function removePluginAction()
         echo $result->getContent();
     }
 
+    public function deactivateConnectProductsAction()
+    {
+        $this->Response()->setHeader('Content-Type', 'application/json; charset=utf-8');
+
+        $request = file_get_contents('php://input');
+
+        $apiRequest = $this->getRestApiRequest();
+        $result = $apiRequest->verifyRequest($request);
+
+        if ($result->isOk()) {
+            Shopware()->Db()->exec(
+                'UPDATE s_articles
+                INNER JOIN s_plugin_connect_items
+                  ON s_plugin_connect_items.article_id = s_articles.id
+                  AND shop_id IS NOT NULL
```

Comment:
get supplier id from request data. and chance condition to AND shop_id  = shopIdFromRequest.
Also alter this to prepared statement since parameters will be involved